### PR TITLE
Elevation range edit

### PR DESCRIPTION
**Pull request by swalker2m**
Adds elevation range editing.

### CODE REVIEW

File: modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala [hunk 1]
```diff
@@ -29,38 +35,78 @@ import lucuma.odb.data.Nullable.NonNull
...
+
+  val ConstraintViolationMessages: List[(String, String)] =
+    List(
+      MissingAirMassConstraint   -> MissingAirMassConstraintMessage,
+      MissingHourAngleConstraint -> MissingHourAngleConstraintMessage
+    )
+
+  def constraintViolationMessage(ex: PostgresErrorException): String =
+    ConstraintViolationMessages
+      .find { case (c, _) => ex.message.contains(c) }
```
Comment (swalker2m) [inline modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:90]: Couldn't see a better way to figure out which constraint was violated.

File: modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala [hunk 2]
```diff
@@ -197,16 +256,42 @@ object ObservationService {
...
+
+        val hourAngle: List[AppliedFragment] =
+          List(
+            e.hourAngle.flatMap(_.minBigDecimal).map(v => upHourAngleMin(v.some)),
+            e.hourAngle.flatMap(_.maxBigDecimal).map(v => upHourAngleMax(v.some))
+          ).flattenOption
+
+        (airMass, hourAngle) match {
+          case (Nil, Nil) => List.empty[AppliedFragment].asRight
+          case (am,  Nil) => (upHourAngleMin(None) :: upHourAngleMax(None) :: am).asRight
```
Comment (swalker2m) [inline modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:290]: when you switch from one kind of elevation constraint to another, you have to wipe out the opposing constraint type

**Review by tpolecat — APPROVED** [review]
:+1:

File: modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala [hunk 3]
```diff
@@ -220,26 +305,33 @@ object ObservationService {
...
-    ): Option[AppliedFragment] =
-      updates(SET).map { us =>
-        void"UPDATE t_observation " |+|
-        void"SET " |+| us.intercalate(void", ") |+| void" " |+|
-        void"WHERE t_observation.c_observation_id IN (" |+| which |+| void")" |+|
-        void"RETURNING t_observation.c_observation_id"
+    ): Either[String, Option[AppliedFragment]] =
+      updates(SET).map {
+        _.map { us =>
+          void"UPDATE t_observation " |+|
```
Comment (cquiroz) [inline modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:330]: Is this skunk syntax?

Comment (swalker2m) [reply modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:330]: Yeah, `void` makes a Skunk `AppliedFragment` out of a bit of SQL that doesn't involve any parameters.  The `sql` interpolator is used when there are query arguments.  At least, that's my understanding.

**Pull request merged**